modules/service/src/main/resources/db/migration/V0995__imaging_science_requirements.sql

```diff
@@ -0,0 +1,150 @@
+  ALTER COLUMN c_science_mode DROP DEFAULT;
+
+-- Rename spec constraint
+-- the exposure time mode is shared between imaging and spectroscopy
+ALTER TABLE t_observation RENAME COLUMN c_spec_exp_time_mode      TO c_exp_time_mode;
+ALTER TABLE t_observation RENAME COLUMN c_spec_signal_to_noise    TO c_etm_signal_to_noise;
+ALTER TABLE t_observation RENAME COLUMN c_spec_signal_to_noise_at TO c_etm_signal_to_noise_at;
+ALTER TABLE t_observation RENAME COLUMN c_spec_exp_time           TO c_etm_exp_time;
+ALTER TABLE t_observation RENAME COLUMN c_spec_exp_count          TO c_etm_exp_count;
+
+-- Add check constraint to ensure c_science_mode is consistent with spectroscopy and imaging fields
+ALTER TABLE t_observation
+  ADD CONSTRAINT science_mode_consistency CHECK (
+    CASE c_science_mode
+      -- If spectroscopy mode, all imaging fields must be NULL
+      WHEN 'spectroscopy' THEN
+        c_img_minimum_fov IS NULL AND
+        c_img_narrow_filters IS NULL AND
+        c_img_broad_filters IS NULL AND
+        c_img_combined_filters IS NULL
+      -- If imaging mode, all spectroscopy fields must be NULL
+      WHEN 'imaging' THEN
+        c_spec_wavelength IS NULL AND
+        c_spec_resolution IS NULL AND
+        c_spec_wavelength_coverage IS NULL AND
+        c_spec_focal_plane IS NULL AND
+        c_spec_focal_plane_angle IS NULL AND
+        c_spec_capability IS NULL
+      -- If science mode is NULL, all science requirement fields must be NULL
+      ELSE
+        c_spec_wavelength IS NULL AND
+        c_spec_resolution IS NULL AND
+        c_spec_wavelength_coverage IS NULL AND
+        c_spec_focal_plane IS NULL AND
+        c_spec_focal_plane_angle IS NULL AND
+        c_spec_capability IS NULL AND
+        c_img_minimum_fov IS NULL AND
+        c_img_narrow_filters IS NULL AND
+        c_img_broad_filters IS NULL AND
+        c_img_combined_filters IS NULL
+    END
+  );
+
+-- Create a trigger to automatically set science mode based on field values
+CREATE OR REPLACE FUNCTION auto_set_science_mode()
+RETURNS TRIGGER AS $$
+BEGIN
+  -- Check if any spectroscopy fields are set
+  IF NEW.c_spec_wavelength IS NOT NULL OR
+     NEW.c_spec_resolution IS NOT NULL OR
+     NEW.c_spec_wavelength_coverage IS NOT NULL OR
+     NEW.c_spec_focal_plane IS NOT NULL OR
+     NEW.c_spec_focal_plane_angle IS NOT NULL OR
+     NEW.c_spec_capability IS NOT NULL THEN
+    -- If spectroscopy fields exist, set mode to spectroscopy
+    NEW.c_science_mode := 'spectroscopy';
+    -- Check if any imaging fields are set
+  ELSIF NEW.c_img_minimum_fov IS NOT NULL OR
+        NEW.c_img_narrow_filters IS NOT NULL OR
+        NEW.c_img_broad_filters IS NOT NULL OR
+        NEW.c_img_combined_filters IS NOT NULL THEN
+    -- If imaging fields exist, set mode to imaging
+    NEW.c_science_mode := 'imaging';
+  ELSE
+    -- If no fields are set, set mode to null
+    NEW.c_science_mode := NULL;
+  END IF;
+
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
We might consider making `c_science_mode` a `GENERATED` column if it is just a function of the other columns like this?